### PR TITLE
Add CloudTasksQueue resource

### PR DESCRIPTION
**Pull request by tysen**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
google_cloud_tasks_queue
```

### CODE REVIEW

File: templates/terraform/examples/queue_basic.tf.erb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+resource "google_cloud_tasks_queue" "default" {
+  name = "<%= ctx[:vars]['name'] %>"
+  location = "us-central1"
+
```
Comment (slevenick) [inline templates/terraform/examples/queue_basic.tf.erb]: Empty line

File: third_party/terraform/tests/resource_cloud_tasks_queue_test.go.erb [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  rate_limits {
+    max_concurrent_dispatches = 4
+    max_dispatches_per_second = 3
+  }
+
+  retry_config {
+    max_attempts = 6
+    max_retry_duration = "5s"
+    max_backoff = "4s"
+    min_backoff = "3s"
```
Comment (slevenick) [inline third_party/terraform/tests/resource_cloud_tasks_queue_test.go.erb]: What happens if the user doesn't specify one of these on the update?

File: templates/terraform/custom_flatten/cloudtasks_queue_appenginerouting.go.erb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+// service, version, and instance are input-only. host is output-only.
+func flatten<%= prefix -%><%= titlelize_property(property) -%>(v interface{}, d *schema.ResourceData) interface{} {
+	if v == nil {
+		return nil
+	}
+	original := v.(map[string]interface{})
+	if len(original) == 0 {
+		return nil
+	}
+  transformed := make(map[string]interface{})
```
Comment (slevenick) [inline templates/terraform/custom_flatten/cloudtasks_queue_appenginerouting.go.erb]: Tabs vs spaces in this file

File: third_party/terraform/tests/resource_cloud_tasks_queue_test.go.erb (hunk 2 above)
Comment (slevenick) [reply third_party/terraform/tests/resource_cloud_tasks_queue_test.go.erb]: For example, they go from this `retry_config` block to:
```
retry_config {
  max_attempts = 10
}
```

Comment (tysen) [reply third_party/terraform/tests/resource_cloud_tasks_queue_test.go.erb]: Good question. Made the fields computed and added a test.

File: products/cloudtasks/terraform.yaml [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      rateLimits: !ruby/object:Overrides::Terraform::PropertyOverride
+        default_from_api: true
+      rateLimits.maxConcurrentDispatches: !ruby/object:Overrides::Terraform::PropertyOverride
+        default_from_api: true
+      rateLimits.maxDispatchesPerSecond: !ruby/object:Overrides::Terraform::PropertyOverride
+        default_from_api: true
+      retryConfig: !ruby/object:Overrides::Terraform::PropertyOverride
+        default_from_api: true
+      retryConfig.maxAttempts: !ruby/object:Overrides::Terraform::PropertyOverride
+        default_from_api: true
```
[the remark this answers is not in the document]
Comment (tysen) [reply products/cloudtasks/terraform.yaml]: It doesn't work if just `retryConfig` is set to computed. Terraform would want the non-computed fields to be unset in that case.

**Comment by tysen** [comment]
Updated the changelog entry and enabled the API in the one CI project I'm aware of.

**Comment by callppatel** [comment]
When this resource will be available for google provider?

**Comment by jmdobry** [comment]
Is there an automatic release where support Cloud Task Queues will become available?

**Comment by tysen** [comment]
It's in 3.1.0

**Comment by jmdobry** [comment]
Will it be back-ported to 2.x?

**Comment by tysen** [comment]
It won't be backported - we don't have the bandwidth to maintain two separate release chains.

**Pull request merged**